// plutosols-greeting/app.js
module.exports = function () {
    var firstName = '';
    var lastName = '';   
    var root = document.getElementById('root');

    this.GetFirstName = function () {
        return firstName;
    };

    this.GetLastName = function () {
        return lastName;
    };

    this.SetGreeting = function (vFirstName, vLastName) {
        firstName = vFirstName;
        lastName = vLastName;
    };

    this.SetFirstName = function (vFirstName) {
        firstName = vFirstName;
    };

    this.SetLastName = function (vLastName) {
        lastName = vLastName;
    };

    this.GetFullName = function () {
        return firstName + ' ' + lastName;
    };

    this.SetName = function (vFirstName, vLastName) {
        firstName = vFirstName;
        lastName = vLastName;
    };

    this.Greeting = function () {
        root.appendChild(this.H1('Welcome to Pluto Solutions'));
        root.appendChild(this.H2('Hello, ' + this.GetFullName()));
        root.appendChild(this.Text('I came from Pluto JavaScript Server'));
    };

    this.ClearRoot = function () {
        root.innerHTML = "";
    };

    this.ClearElement = function (id) {
        document.getElementById(id).innerHTML = "";
    };

    this.AddElementTo = function (childId, mainId) {
        document.getElementById(mainId).appendChild(childId);
    };

    this.AddElement = function (id) {
        root.appendChild(id);
    };

    this.Div = function (text) {
        var element = document.createElement('div');
        element.innerHTML = text;
        return element;
    };

    this.H1 = function (text) {
        var element = document.createElement('h1');
        element.innerHTML = text;
        return element;
    };

    this.H2 = function (text) {
        var element = document.createElement('h2');
        element.innerHTML = text;
        return element;
    };

    this.H3 = function (text) {
        var element = document.createElement('h3');
        element.innerHTML = text;
        return element;
    };

    this.H4 = function (text) {
        var element = document.createElement('h4');
        element.innerHTML = text;
        return element;
    };

    this.H5 = function (text) {
        var element = document.createElement('h5');
        element.innerHTML = text;
        return element;
    };

    this.Text = function (text) {
        var element = document.createElement('p');
        element.innerHTML = text;
        return element;
    };

    this.Hr = function () {
        var element = document.createElement("hr");
        return element;
    };

    this.Br = function () {
        var element = document.createElement("br");
        return element;
    };
}
